### DISCUSSION TITLE
Nvidia Passthrough

### DISCUSSION THREAD

**Discussion by Deekerman** [tteck/Proxmox#579] [Ideas]
Would be great to have a script that gets everything installed for nvidia on the node then another script to allow it into different LXC or VM. Hopefully this makes sense. I have a 1080 in my server just sitting there since I M DUM.

Thanks!

**Comment by tteck** [comment]
Looking at this

**Reply by lexiismadd** [reply]
This would be awesome if you do come up with it for us noobs

**Reply by MacRimi** [reply]
Hello tteck, can you use my manual if it helps you
https://github.com/MacRimi/manuales/blob/main/NVIDIA/nvidia_proxmox.md

**Reply by lexiismadd** [reply]
will your instructions work with consumer-grade GPUs?

**Reply by MacRimi** [reply]
yes of course

**Reply by mnaoswr** [reply]
@MacRimi I followed your guide but in the lxc the /dev/nv* belongs to nobody. Can you please help?
![grafik](https://github.com/tteck/Proxmox/assets/170001571/188da0fb-6857-44eb-9f02-58773097dd59)

on the host it is configured correctly

**Reply by MacRimi** [reply]
use a privileged LCX

**Reply by MoldyTaint** [reply]
This is my current LXC conf. It is not privileged and has working GPU passthrough
![image](https://github.com/tteck/Proxmox/assets/98976440/2241f9ea-cfad-449c-a511-3f420aa58ffb)

**Comment by MacWeber** [comment]
Why not GPU brand agnostic? Make it for AMD and Intel too ;)

**Reply by Deekerman** [reply]
I can't speak to Intel but from what I was told for AMD its rather simple. Nvidia is more complex. So i dont know if you could make one brand agnostic. 

But if possible sounds great.

**Comment by fguisso** [comment]
I don't know if this works, but there is a script [here](https://github.com/ivanhao/pvetools/blob/d10b811faf80167b12c1f165db34aaaf80e201fc/pvetools.sh#L1944) that seems very simple to use.

I'm using Ryzen 5600H with Radeon(AMD Cezane) and I can't pass through anything, crash my server many times. If anyone makes this work, please, explain here.

**Reply by MoldyTaint** [reply]
I've had issues before too. I finally bit the bullet reinstalling over and over trying to make a step by step that's easy to follow and get it working nearly every time.

`##POST INSTALL SCRIPT##

bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/post-pve-install.sh)"

##KERNEL CLEAN##

bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/kernel-clean.sh)"

##PROCESSOR MICRO CODE##

bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/microcode.sh)"

##CHECK MICRO CODE##

journalctl -k | grep -E "microcode: microcode"

##INSTALL PVE HEADERS##

apt install pve-headers



##INSTALL NVIDIA DRIVERS##

Blacklist Nouveau Drivers
"echo "blacklist nouveau" >> /etc/modprobe.d/blacklist.conf"
REBOOT
Go here: https://www.nvidia.com/Download/index.aspx
Input GPU information
"wget (PUT RUN FILE HERE)"
"apt install gcc && apt install make"
"sh ./(nvidia run file)"
Say Yes to everything

##LXC GPU DRIVER INSTALL##

Go here: https://www.nvidia.com/Download/index.aspx
Input GPU information
"wget (SAME EXACT RUN FILE FROM BEFORE)"
"sh ./(downloadedfile --no-kernel-module"
`

**Reply by kzaoaai** [reply]
unfortunately, this is not working for proxmox 8 (kernel 6.2)

**Reply by MoldyTaint** [reply]
Haven't had an issue with Nvidia Passthrough on 8.0. Been using it fluently without any issues since 8.0 dropped.

**Reply by kzaoaai** [reply]
weird, my plex lxc will throw an error and not use hardware transcoding even though nvidia-smi shows the card.

**Reply by MoldyTaint** [reply]
My servers were taken offline about 2 weeks ago while we move into our new place. As of two weeks ago I had 100% working Hardware transcoding through Plex and Jellyfin. I have a step by step down that I follow everytime that gets it working. When I pull it up later I'll send it over to ya!

**Reply by kzaoaai** [reply]
that would be great, thanks!

**Reply by kzaoaai** [reply]
It's working again. The key (at least for me) was to uninstall the driver, restart proxmox, and install NVIDIA drivers again after the proxmox upgrade.

**Reply by sco0by2k** [reply]
@MoldyTaint I have been trying to get the Plex LXC to work with Nvidia Transcoding for the longest time and your guide works! Every other guide I tried is either wrong, or has unnecessary steps. 

The only thing I had to do differently was to also blacklist the nvidiafb drivers, and run "update-initramfs -u" but this may have been due to my previous attempts to get it working.

nvidia-smi now shows the card and when I play a video in Plex it shows as Transcode (hw), and the LXC CPU usage is around 20% even when playing huge 4k files.

Thanks so much for the simple but effective guide, you should post it as a new post to make it easier for anyone else struggling to find it! :D

**Reply by RuiRC** [reply]
@MoldyTaint Your solution is still valid to this day, thank you very much!

**Reply by manuelkamp** [reply]
i tried this for localai (running in docker) and was not successful. maybe it would help to have localai as lxc install...?
